examples/c/profile.c

```diff
@@ -113,6 +114,85 @@ static void show_help(const char *progname)
 	printf("Usage: %s [-f <frequency>] [-h]\n", progname);
 }
 
+// copy from https://github.com/libbpf/libbpf/blob/v1.2.0/src/libbpf.c#L12420
+int bootstrap_parse_cpu_mask_str(const char *s, bool **mask, int *mask_sz)
+{
+	int err = 0, n, len, start, end = -1;
+	bool *tmp;
+
+	*mask = NULL;
+	*mask_sz = 0;
+
+	/* Each sub string separated by ',' has format \d+-\d+ or \d+ */
+	while (*s) {
+		if (*s == ',' || *s == '\n') {
+			s++;
+			continue;
+		}
+		n = sscanf(s, "%d%n-%d%n", &start, &len, &end, &len);
+		if (n <= 0 || n > 2) {
+			fprintf(stderr, "Failed to get CPU range %s: %d\n", s, n);
+			err = -EINVAL;
+			goto cleanup;
+		} else if (n == 1) {
+			end = start;
+		}
+		if (start < 0 || start > end) {
+			fprintf(stderr, "Invalid CPU range [%d,%d] in %s\n",
+				start, end, s);
+			err = -EINVAL;
+			goto cleanup;
+		}
+		tmp = realloc(*mask, end + 1);
+		if (!tmp) {
+			err = -ENOMEM;
+			goto cleanup;
+		}
+		*mask = tmp;
+		memset(tmp + *mask_sz, 0, start - *mask_sz);
+		memset(tmp + start, 1, end - start + 1);
+		*mask_sz = end + 1;
+		s += len;
+	}
+	if (!*mask_sz) {
+		fprintf(stderr, "Empty CPU range\n");
+		return -EINVAL;
+	}
+	return 0;
+cleanup:
+	free(*mask);
+	*mask = NULL;
+	return err;
+}
+
+// copy from https://github.com/libbpf/libbpf/blob/v1.2.0/src/libbpf.c#L12470
+int bootstrap_parse_cpu_mask_file(const char *fcpu, bool **mask, int *mask_sz)
```

Comment:
make it static? and then you don't need to prepend `bootstrat_` prefix, just leave original name?